src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -97,6 +99,15 @@ public void setPersons(List<Person> persons) {
         internalList.setAll(persons);
     }
 
+    /**
+     * Sets the given person with {@code name} in the list as a {@code buyer}.
+     * {@code name} exists uniquely in the address book.
+     */
+    public void setContactRole(Name name, Buyer buyer) {
+        FilteredList<Person> matchingNamePersons = internalList.filtered(x -> x.getName().equals(name));
+        matchingNamePersons.get(0).setBuyer(buyer);
```

Comment:
hmm maybe we should match by relative index instead, like what is currently being done in AB3